airbyte-to-flow/src/interceptors/fix_document_schema.rs

```diff
@@ -99,41 +100,56 @@ pub fn fix_document_schema_keys(mut doc: serde_json::Value, key_ptrs: Vec<String
     Ok(doc)
 }
 
-pub fn fix_nonstandard_jsonschema_attributes(schema: &mut serde_json::Value) {
+pub fn traverse_jsonschema<F>(schema: &mut serde_json::Value, f: &mut F)
+where F: FnMut(&mut Map<String, serde_json::Value>) -> () {
     match schema {
         serde_json::Value::Object(map) => {
-            // airbyte sometimes hides some fields from their config but keeps them
-            // for backward compatibility
-            map.remove("airbyte_hidden");
-
-            // "group" is an attribute airbyte uses internally
-            map.remove("group");
-
-            // a mapping from a jsonschema type to an internal airbyte type
-            map.remove("airbyte_type");
-
+            f(map);
             // keys under properties are schemas, so we need to run on those as well
             map.get_mut("properties").map(|props| {
                 match props {
                     serde_json::Value::Object(inner_map) => {
                         inner_map.values_mut().for_each(|v| {
-                            fix_nonstandard_jsonschema_attributes(v)
+                            traverse_jsonschema(v, f);
                         });
                     },
                     _ => ()
                 }
             });
-            map.get_mut("items").map(fix_nonstandard_jsonschema_attributes);
+            map.get_mut("items").map(|item| traverse_jsonschema(item, f));
         }
         _ => ()
     }
 }
 
+pub fn fix_nonstandard_jsonschema_attributes(schema: &mut serde_json::Value) {
```

Comment:
Nit: I might try refactoring this to call `traverse_jsonschema(f)` the interceptor, and then these functions would not call `traverse_jsonschema`. I think that would make it easier to add more transformation functions in the future (maybe you can iterate on a list of functions? like `[f1, f2, f3].iter().forEach(|f| traverse_jsonschema(f))` - I don't know how this actually works in rust though), but not a big deal.